docs/getting_started/filtering.md

```diff
@@ -0,0 +1,108 @@
+This guide will briefly outline how to parse OPTIMADE filter strings into database or API-specific queries.
+
+## Parsing OPTIMADE filter strings
+
+Example use:
+
+```python
+from optimade.filterparser import LarkParser
+
+p = LarkParser(version=(1, 0, 0))
+tree = p.parse("nelements<3")
+print(tree)
+```
+
+```shell
+Tree('filter', [Tree('expression', [Tree('expression_clause', [Tree('expression_phrase', [Tree('comparison', [Tree('property_first_comparison', [Tree('property', [Token('IDENTIFIER', 'nelements')]), Tree('value_op_rhs', [Token('OPERATOR', '<'), Tree('value', [Tree('number', [Token('SIGNED_INT', '3')])])])])])])])])])
+```
+
+```python
+print(tree.pretty())
+```
+
+```shell
+filter
+  expression
+    expression_clause
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	nelements
+            value_op_rhs
+              <
+              value
+                number	3
+```
+
+```python
+tree = p.parse('_mp_bandgap > 5.0 AND _cod_molecular_weight < 350')
+print(tree.pretty())
+```
+
+```shell
+filter
+  expression
+    expression_clause
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	_mp_bandgap
+            value_op_rhs
+              >
+              value
+                number	5.0
+      expression_phrase
+        comparison
+          property_first_comparison
+            property	_cod_molecular_weight
+            value_op_rhs
+              <
+              value
+                number	350
+```
+
+## Flow for parsing a user-supplied filter and converting to a backend query
+
+`optimade.filterparser.LarkParser` will take user input to generate a `lark.Tree` and feed that to a `lark.Transformer`.
+For example, `optimade.filtertransformers.mongo.MongoTransformer` will turn the tree into something useful for a MongoDB backend:
+
+```python
+# Example: Converting to MongoDB Query Syntax
+from optimade.filtertransformers.mongo import MongoTransformer
+
+transformer = MongoTransformer()
+
+tree = p.parse('_mp_bandgap > 5.0 AND _cod_molecular_weight < 350')
+query = transformer.transform(tree)
+print(query)
+```
+
+```json
+{
+    "$and": [
+        {"_mp_bandgap": {"$gt": 5.0}},
+        {"_cod_molecular_weight": {"$lt": 350.0}}
+    ]
+}
+```
+
+
+## Developing new filter transformers
+
+In order to support a new backend, you will need to create a new filter transformer that inherits from the [`BaseTransformer`][optimade.filtertransformers.base_transformer.BaseTransformer].
+This transformer will need to override the methods that match the particular grammatical constructs in the Lark grammar in order to construct a query.
+Two examples can be found for MongoDB ([`MongoTransformer`][optimade.filtertransformers.mongo.MongoTransformer]) and Elasticsearch ([`ElasticTransformer`][optimade.filtertransformers.elasticsearch.ElasticTransformer]).
```

Comment:
When I first read this sentence, I interpreted it as if there were two transformers for mongo DB.
Perhaps this is clearer.
```suggestion
Two examples can be found within 'optimade-python-tools', one for MongoDB ([`MongoTransformer`][optimade.filtertransformers.mongo.MongoTransformer]) and one for Elasticsearch ([`ElasticTransformer`][optimade.filtertransformers.elasticsearch.ElasticTransformer]).
```